### PR TITLE
1187: Expert is added to an off-chain whitelist for their project application

### PR DESCRIPTION
**Pull request by nagmak**
### What changes were made?

- create a proposal index to access a proposal by ID
- user permissions - ensure that only the client that is the owner of the contract can see the applicants list and edit the bounty
- associate proposals with a contract as an off-chain whitelist and also add proposals to the user who created the contract

### Is there any background info about the change?

part of the application and approval flow

### Are there any state changes? Mention key ones (if any)
NA

### Screenshots or Gifs✨

NA

### Any new dependencies? Why were they added?
NA

### Shortcut Link

https://app.shortcut.com/twali/story/1102/expert-is-added-to-an-off-chain-whitelist-for-their-project-application

### CODE REVIEW

File: data/dynamodb.js [hunk 1]
```diff
@@ -627,67 +648,92 @@ module.exports = {
...
+            ":application_status": applicationStatus,
+            ":expert_profile_img": expertProfileImg ? expertProfileImg : null,
+            ":isFavorited": isFavorited ? isFavorited : null,
+            ":rejection_text": rejectionText ? rejectionText : null,
+            ":contract_id": contractID
+          },
+        })
+        .promise()
+        .then((data) => {
+          console.log('createProposal: ', data)
```
Comment (NickGonzalez04) [inline data/dynamodb.js:682]: 🔫 phew phew

File: data/dynamodb.js [hunk 2]
```diff
@@ -627,67 +648,92 @@ module.exports = {
...
+    /**
+   * @desc Returns a list of proposals by contract ID
+   * @dev Can implement a value check in the near future.
+   * @returns Returns a boolean value
+   */
+     getProposals: async (bounty) => {
+      console.log('HELLO THERE');
+      console.log('getProposals!!!!!!!');
+      console.log(bounty);
+      await getDynamoDBClient()
```
Comment (NickGonzalez04) [inline data/dynamodb.js:695]: 🔫 🔫 🔫

**Review by NickGonzalez04 — APPROVED** [review]
Over here causing havoc for console.logs for no reason!

**Review by hone1er — APPROVED** [review]
Noice

**Pull request merged**